client/control/components/RecipeList.js

```diff
@@ -64,14 +64,61 @@ SwitchFilter.propTypes = {
 };
 
 class DisconnectedRecipeList extends React.Component {
-  propTypes = {
-    dispatch: pt.func.isRequired,
-    isFetching: pt.bool.isRequired,
-    recipeListNeedsFetch: pt.bool.isRequired,
-    recipe: pt.object.isRequired,
-    recipes: pt.array.isRequired,
+  /**
+   * Recipe metadata properties and associated labels to display
+   * @type {Object}
+   */
+  static ActionMetaData = {
+    'show-heartbeat': {
+      // prop name : displayed label
+      surveyId: 'Survey ID',
+    },
+  };
```

Comment:
What do you think about a slightly more procedural way of doing declaring this data, like this:

```js
static recipeMetadata = {
  'show-heartbeat': recipe => [{ label: 'Survey ID', value: recipe.surveyId }],  
};
```

I think this is a little more direct and easier to understand, and also a bit more flexible if we want to pull more complex data out of recipes (such as values in the filter expression).
```